internal/controllers/nmc_reconciler.go

```diff
@@ -749,6 +748,57 @@ func setWorkerContainerArgs(pod *v1.Pod, args []string) error {
 	return nil
 }
 
+func setWorkerSofdepConfig(pod *v1.Pod, modulesLoadingOrder []string) {
+	softdepAnnotationValue := getModulesOrderAnnotationValue(modulesLoadingOrder)
+	setWorkerPodAnnotation(pod, "modules-order", softdepAnnotationValue)
+
+	softdepVolume := v1.Volume{
+		Name: "modules-order",
+		VolumeSource: v1.VolumeSource{
+			DownwardAPI: &v1.DownwardAPIVolumeSource{
+				Items: []v1.DownwardAPIVolumeFile{
+					{
+						Path:     "softdep.conf",
+						FieldRef: &v1.ObjectFieldSelector{FieldPath: "metadata.annotations['modules-order']"},
+					},
+				},
+			},
+		},
+	}
+	softDepVolumeMount := v1.VolumeMount{
+		Name:      "modules-order",
+		ReadOnly:  true,
+		MountPath: "/etc/modprobe.d",
+	}
+	pod.Spec.Volumes = append(pod.Spec.Volumes, softdepVolume)
+	pod.Spec.Containers[0].VolumeMounts = append(pod.Spec.Containers[0].VolumeMounts, softDepVolumeMount)
+}
+
+func getModulesOrderAnnotationValue(modulesNames []string) string {
+	softDepData := ""
+	for i := 0; i < len(modulesNames); i++ {
+		if i == len(modulesNames)-1 {
+			break
+		}
+		line := prepareSoftDepLine(modulesNames[i], modulesNames[i+1])
+		softDepData = softDepData + line
+	}
+	return softDepData
+}
+
+func prepareSoftDepLine(dependendModuleName, moduleName string) string {
+	return fmt.Sprintf("softdep %s pre: %s\n", dependendModuleName, moduleName)
+}
```

Comment:
Why do we need this function?

Reply:
i added it for clearer code, so that we can see what module goes where in the file, but i removed it when moved to builder